cmd/root.go

```diff
@@ -0,0 +1,198 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	pathpkg "path"
+	"strings"
+
+	"cloud.google.com/go/spanner"
+	"github.com/spf13/cobra"
+	"go.mercari.io/yo/generator"
+	"go.mercari.io/yo/internal"
+	"go.mercari.io/yo/loaders"
+)
+
+const (
+	defaultSuffix = ".yo.go"
+	exampleUsage  = `
+  # Generate models under models directory
+  yo $PROJECT_NAME $INSTANCE_NAME $DATABASE_NAME -o models
+
+  # Generate models under models directory with custom types
+  yo $PROJECT_NAME $INSTANCE_NAME $DATABASE_NAME -o models --custom-types-file custom_column_types.yml
+`
+)
+
+var (
+	rootOpts = internal.ArgType{}
+	rootCmd  = &cobra.Command{
+		Use:   "yo PROJECT_NAME INSTANCE_NAME DATABASE_NAME",
+		Short: "yo is a command-line tool to generate Go code for Google Cloud Spanner.",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 3 {
+				return fmt.Errorf("must specify 3 arguments")
+			}
+			return nil
+		},
+		Example: strings.Trim(exampleUsage, "\n"),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := processArgs(&rootOpts, args); err != nil {
+				return err
+			}
+
+			spannerClient, err := connectSpanner(&rootOpts)
```

Comment:
I fixed codes based on [cobra example](https://github.com/spf13/cobra#getting-started).
Can't add `create-template` subcommand without moving main function to rootCmd.RunE.

Reply:
nice. We can refactor later.